### PR TITLE
fix(session): sanitize dots in tmux window names + regression tests

### PR DESCRIPTION
**Pull request by filipexyz**
## Summary

- **fix(session):** Sanitize `.` → `-` in tmux window names to prevent "can't find pane" errors
- **test(session):** 8 regression tests for `sanitizeWindowName` (pure function, no mock.module)
- **chore(version):** bump to 3.260310.5

## Problem

Running `genie <folder.with.dots>` caused:
```
Error: Failed to execute tmux command: Command failed: tmux select-window -t 'genie:folder.name'
can't find pane: name
```

tmux interprets `.` as a pane separator in target strings (`session:window.pane`), so dotted names get parsed as window + pane instead of just window name.

## Fix

Added `sanitizeWindowName()` in `session.ts` that replaces all dots with dashes before passing names to tmux. Applied in `deriveWindowName()` which handles all three name sources: explicit `--team`, folder basename, and resolved window name.

## Test plan

- [x] Dotted names sanitized (e.g. `my.app` → `my-app`)
- [x] Multiple dots handled (`my.cool.app` → `my-cool-app`)
- [x] Names without dots unchanged
- [x] Edge cases: leading/trailing dots, consecutive dots, empty string, only dots
- [x] 535 tests pass, 0 fail

## Summary by CodeRabbit

* **Bug Fixes**
  * Window and team names are now sanitized to avoid tmux parsing conflicts and naming collisions.

* **Chores**
  * Bumped plugin and package versions to 3.260310.5.

* **Tests**
  * Expanded tests covering window-name sanitization, edge cases, and session behavior.

* **Refactor**
  * Improved idle-worker detection and suspension for more reliable idle handling.

### CODE REVIEW

File: src/genie-commands/__tests__/session.test.ts [hunk 1]
```diff
@@ -115,3 +115,53 @@ describe('getAgentsSystemPrompt', () => {
...
+  });
+
+  test('handles consecutive dots', () => {
+    expect(sanitizeWindowName('a..b')).toBe('a--b');
+  });
+
+  test('handles directory basename with dot (simulates basename of /home/user/ravi.bot)', () => {
+    // In production, basename() is called before sanitizeWindowName,
+    // so we test the basename result directly.
+    const { basename } = require('node:path');
```
[the remark this answers is not in the document]
Comment (filipexyz) [reply src/genie-commands/__tests__/session.test.ts]: Fixed in 1214ca1d — `require('node:path')` replaced with top-level ESM import `import { basename, join } from 'node:path'`. Good catch. ✅

File: src/genie-commands/session.ts [hunk 2]
```diff
@@ -235,11 +235,20 @@ async function focusTeamWindow(
...
+  return name.replace(/\./g, '-');
+}
+
 async function deriveWindowName(sessionName: string, workspaceDir: string, team?: string): Promise<string> {
-  if (team) return team;
+  if (team) return sanitizeWindowName(team);
   const existingSession = await tmux.findSessionByName(sessionName);
-  if (existingSession) return resolveWindowName(sessionName, workspaceDir);
-  return basename(workspaceDir);
+  if (existingSession) return sanitizeWindowName(await resolveWindowName(sessionName, workspaceDir));
```
[the remark this answers is not in the document]
Comment (filipexyz) [reply src/genie-commands/session.ts:253]: This is a theoretical edge case that doesn't apply in practice. Before this fix, dotted window names like `ravi.bot` caused tmux to crash with `can't find pane: bot` — they never worked at all. There are no pre-existing windows with dots to orphan or collide with.

The `a.b` → `a-b` collision scenario (where someone has both `a.b/` and `a-b/` as workspace directories) is extremely unlikely, and even if it occurred, the existing `resolveWindowName` hash-based disambiguation handles it — both would get unique hashed suffixes.

Marking as acknowledged — no code change needed for a non-existent edge case.

File: src/genie-commands/session.ts [hunk 3]
```diff
@@ -235,11 +235,20 @@ async function focusTeamWindow(
...
+  return sanitizeWindowName(basename(workspaceDir));
 }
```
Comment (filipexyz) [inline src/genie-commands/session.ts:255]: Same reasoning as above — before this fix, dotted window names caused immediate tmux failures (`can't find pane: <suffix after dot>`). There are no pre-existing working windows with dots in their names to worry about.

The sanitization happens at `deriveWindowName` level, which is the single entry point for all window name resolution. `resolveWindowName` and `findWindowByName` operate on already-sanitized names since `deriveWindowName` calls `sanitizeWindowName()` on all return paths. New windows are created with sanitized names, and subsequent lookups find them correctly.

No action needed.

File: src/lib/idle-timeout.test.ts [hunk 4]
```diff
@@ -10,20 +10,59 @@ let mockExecuteTmuxCalls: string[] = [];
...
+  createWindow: async () => null,
+  findWindowByName: async () => null,
+  ensureTeamWindow: async () => ({ windowId: '', created: false }),
+  applyPaneColor: async () => {},
+  killSession: async () => {},
+  killWindow: async () => {},
+  killWindowQualified: async () => {},
+  killPane: async () => {},
+  executeCommand: async () => ({ success: false, output: '' }),
+  runCommandSync: async () => ({ success: false, output: '' }),
```
[the remark this answers is not in the document]
Comment (filipexyz) [reply src/lib/idle-timeout.test.ts]: Moot — commit 21ef329 replaced mock.module entirely with dependency injection, so these stubs no longer exist.

File: src/genie-commands/session.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (filipexyz) [reply src/genie-commands/session.ts:253]: Same response as before — dotted window names caused immediate tmux crashes pre-fix. No pre-existing working windows with dots exist. The hash-based disambiguation in `resolveWindowName` handles the theoretical `foo.bar` / `foo-bar` collision case.

File: src/genie-commands/session.ts [hunk 5]
```diff
@@ -235,11 +238,20 @@ async function focusTeamWindow(
...
```
[the remark this answers is not in the document]
Comment (filipexyz) [reply src/genie-commands/session.ts:255]: Valid catch! Fixed — `team-auto-spawn.ts` now calls `sanitizeWindowName(teamName)` before passing to `ensureTeamWindow` and before constructing the tmux target string. Incoming in next commit.

File: src/lib/team-auto-spawn.ts [hunk 6]
```diff
@@ -103,13 +104,14 @@ export async function ensureTeamLead(teamName: string, workingDir: string): Prom
   // Ensure tmux session exists
   const session = await ensureSession();
 
-  // Create team window
-  const teamWindow = await tmux.ensureTeamWindow(session, teamName, workingDir);
+  // Create team window (sanitize dots — tmux interprets '.' as pane separator)
+  const windowName = sanitizeWindowName(teamName);
+  const teamWindow = await tmux.ensureTeamWindow(session, windowName, workingDir);
```
[the remark this answers is not in the document]
Comment (filipexyz) [reply src/lib/team-auto-spawn.ts:109]: Good catch — fixed in 2ab1cd05. Both return paths now use `sanitizeWindowName` (the same function used to create the window), so the returned name always matches the actual tmux window.
